server/src/test/java/io/unitycatalog/server/base/catalog/BaseCatalogCRUDTest.java

```diff
@@ -1,35 +1,36 @@
 package io.unitycatalog.server.base.catalog;
 
 import static io.unitycatalog.server.utils.TestUtils.*;
+import static org.assertj.core.api.Assertions.assertThat;
 
 import io.unitycatalog.client.ApiException;
 import io.unitycatalog.client.model.CatalogInfo;
 import io.unitycatalog.client.model.CreateCatalog;
 import io.unitycatalog.client.model.UpdateCatalog;
 import io.unitycatalog.server.base.BaseCRUDTest;
+import org.junit.Test;
+
 import java.util.List;
 import java.util.Objects;
 import java.util.Optional;
-import org.junit.*;
 
 public abstract class BaseCatalogCRUDTest extends BaseCRUDTest {
 
   protected void assertCatalog(CatalogInfo catalogInfo, String name, String comment) {
-    Assert.assertEquals(name, catalogInfo.getName());
-    Assert.assertEquals(comment, catalogInfo.getComment());
-    Assert.assertNotNull(catalogInfo.getCreatedAt());
+    assertThat(catalogInfo.getName()).isEqualTo(name);
```

Comment:
The [tests seem to be failing](https://github.com/unitycatalog/unitycatalog/actions/runs/9889636546/job/27316141620?pr=190#step:4:2335) at this assertion? why is that?

Reply:
@vikrantpuppala hmm, that's werd to me as well, initially I thought it depends on the cli implementation because the error log is related to the cli:
>[error] Test io.unitycatalog.**cli.catalog.CliCatalogCRUDTest**.testCatalogCRUD[0] failed: java.lang.NullPointerException: null.

But when I tried to find the root cause, I found 2 strange things:
(1) It's fine if we run testOnly: `build/sbt 'testOnly io.unitycatalog.cli.catalog.CliCatalogOperations'` but error if run with: `build/sbt test`.
(2) `NullPointerException` (same as error above) also threw at `create catalog`'s test-case when comment // [this line](https://github.com/unitycatalog/unitycatalog/blob/ed2ad4af6fbd663e45a0271c2a750d9e2acab9eb/examples/cli/src/test/java/io/unitycatalog/cli/TestUtils.java#L23) (just testing, I don't quite understand the intention of this error handling). But it doesn't work with the new added `update catalog`'s test-case.

I will dig deeper into this issue when I have more time this weekend.

Reply:
This seems to be because of updating properties not implemented for the CLI combined with how the CLI validates input; specifically how no-ops are handled. The CLI sees [here](https://github.com/unitycatalog/unitycatalog/blob/bb8dd4fddee8390043e4133c767165d0baa1050b/examples/cli/src/main/java/io/unitycatalog/cli/CatalogCli.java#L77-L85) that there are no supported optional parameters so it does not return the info about the existing catalog and instead prints a help message. If you comment that part out the CLI passes the command to the server which returns the catalog info and the test then passes.

You can work around this by getting the catalog info after the update with `getCatalog` separately like is done for the empty update test [here](https://github.com/unitycatalog/unitycatalog/blob/bb8dd4fddee8390043e4133c767165d0baa1050b/server/src/test/java/io/unitycatalog/server/base/catalog/BaseCatalogCRUDTest.java#L60-L61). So the test for updating properties could be e.g.:

    // Update catalog properties without updating name and comment
    System.out.println("Testing update catalog: changing properties..");
    UpdateCatalog updateCatalog3 = new UpdateCatalog().properties(NEW_PROPERTIES);
    CatalogInfo updatedCatalogInfo3 =
        catalogOperations.updateCatalog(CATALOG_NEW_NAME, updateCatalog3);
    CatalogInfo catalogInfo4 = catalogOperations.getCatalog(CATALOG_NEW_NAME);
    assertCatalog(catalogInfo4, CATALOG_NEW_NAME, CATALOG_NEW_COMMENT);